### PR TITLE
Improving the Security concepts overviews

### PR DESCRIPTION
**Pull request by msfussell**
This PR address the following issues
- https://github.com/dapr/docs/issues/1832
- https://github.com/dapr/docs/issues/2088
- https://github.com/dapr/docs/issues/1927

It also reworks the overall structure of the security concepts overview, adding new diagrams and providing a comprehensive set of links to all the security features in Dapr, organized under different headings.

### CODE REVIEW

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+type: docs
+title: "Service invocation across namespaces"
+linkTitle: "Service invocation namespaces"
+weight: 1000
+description: "Call between services deployed to different namespaces"
+---
+
+This article describes how you can call between services deployed to different namespaces. By default, you can invoke services within the *same* namespaces by simply referencing the app ID (`nodeapp`):
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md]:
```suggestion
This article describes how you can call between services deployed to different namespaces. By default, you can invoke services within the *same* namespace by simply referencing the app ID (`nodeapp`):
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+```sh
+localhost:3500/v1.0/invoke/nodeapp/method/neworder
+```
+Service invocation also supports calls across namespaces. On all supported hosting platforms, Dapr app IDs conform to a valid FQDN format that includes the target namespace. You can therefore specify both the app ID (`nodeapp`) in addition to the namespace the app runs in (`production`). For example to call the `neworder` method on the `nodeapp`, in the `production` namespace would be like this;
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md]:
```suggestion
Service invocation also supports calls across namespaces. On all supported hosting platforms, Dapr app IDs conform to a valid FQDN format that includes the target namespace. You can therefore specify both the app ID (`nodeapp`) in addition to the namespace the app runs in (`production`). For example to call the `neworder` method on the `nodeapp`, in the `production` namespace would look like this:
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+```sh
+localhost:3500/v1.0/invoke/nodeapp.production/method/neworder
+```
+
+When using service invocation to call an application in a namespace you  qualify it with the namespace. This is especially useful in cross namespace calls in a Kubernetes cluster. As another example, calling the `ping` method on `myapp` which is scoped to the `production` namespace would be like this;
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md]:
```suggestion
When using service invocation to call an application in a namespace you qualify it with the namespace. This is especially useful in cross namespace calls in a Kubernetes cluster. As another example, calling the `ping` method on `myapp` which is scoped to the `production` namespace would look like this:
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+```bash
+https://localhost:3500/v1.0/invoke/myapp.production/method/ping
+```
+
+And calling the same `ping` method using a curl command from an external DNS address, in this case `api.demo.dapr.team` and supplying the Dapr API token for authentication would be like this;
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md]:
```suggestion
And calling the same `ping` method using a curl command from an external DNS address, in this case `api.demo.dapr.team` and supplying the Dapr API token for authentication would look like this:
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md (hunk 1 above)
Comment (hhunter-ms) [reply daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md]:
```suggestion
In this article, you'll learn how you can call between services deployed to different namespaces. By default, service invocation supports invoking services within the *same* namespace by simply referencing the app ID (`nodeapp`):
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md (hunk 2 above)
Comment (hhunter-ms) [reply daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md]:
```suggestion
Service invocation also supports calls across namespaces. On all supported hosting platforms, Dapr app IDs conform to a valid FQDN format that includes the target namespace. You can specify both:

- The app ID (`nodeapp`), and
- The namespace the app runs in (`production`). 

**Example 1**

Call the `neworder` method on the `nodeapp` in the `production` namespace:
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md (hunk 3 above)
Comment (hhunter-ms) [reply daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md]:
```suggestion
When calling an application in a namespace using service invocation, you qualify it with the namespace. This proves useful in cross-namespace calls in a Kubernetes cluster.

**Example 2**

Call the `ping` method on `myapp` scoped to the `production` namespace:
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md (hunk 4 above)
Comment (hhunter-ms) [reply daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md]:
```suggestion
**Example 3**

Call the same `ping` method as example 2 using a curl command from an external DNS address (in this case, `api.demo.dapr.team`) and supply the Dapr API token for authentication:
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-overview.md [hunk 5]
```diff
@@ -39,23 +39,9 @@ The diagram below is an overview of how Dapr's service invocation works.
...
-Service invocation also supports calls across namespaces. On all supported hosting platforms, Dapr app IDs conform to a valid FQDN format that includes the target namespace.
-
-Users can specify both the app ID (`nodeapp`) in addition to the namespace the app runs in (`production`):
-
-```sh
-localhost:3500/v1.0/invoke/nodeapp.production/method/neworder
-```
-
-This is especially useful in cross namespace calls in a Kubernetes cluster.
+Applications can be scoped to namespaces for deployment and security and you can call between services deployed to different namespaces. For more information read the [Service invocation across namespaces]({{< ref "service-invocation-namespaces.md" >}}) article.
```
Comment (hhunter-ms) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-overview.md]:
```suggestion
Applications can be scoped to namespaces for deployment and security, and you can call between services deployed to different namespaces. For more information, read the [Service invocation across namespaces]({{< ref "service-invocation-namespaces.md" >}}) article.
```

File: daprdocs/content/en/operations/components/component-scopes.md [hunk 6]
```diff
@@ -72,29 +72,9 @@ In this example, the Redis component is only accessible to Dapr instances runnin
...
-```
-curl -i -d '{ "message": "hello" }' \
-     -H "Content-type: application/json" \
-     -H "dapr-api-token: ${API_TOKEN}" \
-     https://api.demo.dapr.team/v1.0/invoke/myapp.production/method/ping
-```
-
-## Using namespaces with pub/sub
-Read [Configure Pub/Sub components with multiple namespaces]({{< ref "pubsub-namespaces.md" >}}) for more information on using namespaces with pub/sub.
+Note: It is possible to have the component YAML applied to namespace "A" *reference* the implementation in namespace "B". For example, a component YAML for Redis in namespace "production-A" can set the Redis host address to point to the Redis instance deployed in namespace "production-B". See [Configure Pub/Sub components with multiple namespaces]({{< ref "pubsub-namespaces.md" >}}) for an example.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-scopes.md]:
```suggestion
{{% alert title="Note" color="primary" %}}
The component YAML applied to namespace "A" can *reference* the implementation in namespace "B". For example, a component YAML for Redis in namespace "production-A" can point the Redis host address to the Redis instance deployed in namespace "production-B". 

See [Configure Pub/Sub components with multiple namespaces]({{< ref "pubsub-namespaces.md" >}}) for an example.
{{% /alert %}}
```

File: daprdocs/content/en/operations/components/component-scopes.md [hunk 7]
```diff
@@ -72,29 +72,9 @@ In this example, the Redis component is only accessible to Dapr instances runnin
...
 
 ## Application access to components with scopes
-
 Developers and operators might want to limit access for one database to a certain application, or a specific set of applications.
 To achieve this, Dapr allows you to specify `scopes` on the component YAML. Application scopes added to a component limit only the applications with specific IDs to be able to use the component.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-scopes.md]:
```suggestion
To achieve this, Dapr allows you to specify `scopes` on the component YAML. Application scopes added to a component limit only the applications with specific IDs from using the component.
```

File: daprdocs/content/en/operations/components/component-scopes.md [hunk 8]
```diff
@@ -72,29 +72,9 @@ In this example, the Redis component is only accessible to Dapr instances runnin
...
-     https://api.demo.dapr.team/v1.0/invoke/myapp.production/method/ping
-```
-
-## Using namespaces with pub/sub
-Read [Configure Pub/Sub components with multiple namespaces]({{< ref "pubsub-namespaces.md" >}}) for more information on using namespaces with pub/sub.
+Note: It is possible to have the component YAML applied to namespace "A" *reference* the implementation in namespace "B". For example, a component YAML for Redis in namespace "production-A" can set the Redis host address to point to the Redis instance deployed in namespace "production-B". See [Configure Pub/Sub components with multiple namespaces]({{< ref "pubsub-namespaces.md" >}}) for an example.
 
 ## Application access to components with scopes
-
 Developers and operators might want to limit access for one database to a certain application, or a specific set of applications.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-scopes.md]:
```suggestion
Developers and operators might want to limit access to one database from a certain application, or a specific set of applications.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
grammar/style pass

**Review by greenie-msft — APPROVED** [review]

**Pull request merged**